android/jni/tun2socks/tun2socks.c

```diff
@@ -294,8 +297,13 @@ static void udp_fd_handler(UdpPcb* udp_pcb, int event) {
         return;
     }
 
+    struct socks_udp_header* header = (struct socks_udp_header*)udp_pcb->udp_recv_buffer;
+    size_t socks_udp_header_len = sizeof(struct socks_udp_header);
+    size_t udp_data_len = recv_bytes - socks_udp_header_len;
```

Comment:
All we know (from the check above) is that `recv_bytes > 0`. Without an additional check that `recv_bytes > socks_udp_header_len`, `udp_data_len` could end up negative or zero, right?


Also, looking above, it looks like `udp_recv` copies up to UDP_MAX_DATAGRAM_BYTES bytes from sockfd into udp_recv_buffer, and returns the number of bytes copied/received. Do we have to take any additional care to zero out the remainder of the buffer? Are there any better abstractions, practices, or tools we should be using with this code in general to make it more memory-safe? (Some static analyzer, fuzzer, etc. to catch memory-safety bugs?)

(Also, it looks like the names of other variables that store a count of bytes all end in `_len`, except for `recv_bytes`. Would something like `udp_datagram_len` be a better name for this than `recv_bytes`? (Btw I don't care much what we standardize on for count variables (n, num, count, len), but I do vote that we standardize on something as a team).)